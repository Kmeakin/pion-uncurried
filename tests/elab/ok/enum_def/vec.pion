enum Nat {
    Zero,
    Succ(n: Nat),
}

enum Vec(T: Type, N: Nat) {
    Nil: Vec(T, Zero),
    Cons(head: T, tail: Vec(T, N)): Vec(T, Succ(N)),
}

let nil = Nil;
let cons = Cons;