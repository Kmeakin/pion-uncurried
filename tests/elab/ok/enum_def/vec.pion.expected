enum Nat: Type {
    Zero: Nat,
    Succ(n: Nat): Nat,
}

enum Vec(T: Type, N: Nat): Type {
    Nil: Vec(T, Zero),
    Cons(head: T, tail: Vec(T, N)): Vec(T, Succ(N)),
}

let nil: fn(T: Type, N: Nat) -> Vec(T, Zero) = Nil;

let cons: fn(T: Type, N: Nat, head: T, tail: Vec(T, N)) -> Vec(T, Succ(N)) = Cons;
